src/NUnitConsole/nunit3-console.tests/ResultReporterTests.cs

```diff
@@ -96,12 +91,26 @@ public void ReportSequenceTest()
         }
 
         [Test]
+        public void CheckReportSettingsTest()
+        {
+            var expected = new [] {
+                "Run Settings",
+                "    Setting1Name: Setting1Value",
+                "    Setting2Name: Setting2Value",
+                ""
+            };
+
+            var actualSettingsReport  = GetReportLines(_reporter.WriteRunSettingsReport);
+            Assert.That(expected, Is.EqualTo(actualSettingsReport));
+        }
+
+        [Test]
         public void SummaryReportTest()
         {
-            var expected = new string[] {
+            var expected = new [] {
```

Comment:
Yes, see my comment that just crossed yours. :-)

C# 3.0 should be our target. We could go higher for code outside the
framework, but since it's all in the same project and repository, that
might be confusing. I'll set a higher standard for the gui, which will be
separate.


On Sat, Oct 31, 2015 at 2:13 PM, Neil Colvin <notifications@github.com>
wrote:

> In src/NUnitConsole/nunit3-console.tests/ResultReporterTests.cs
> <https://github.com/nunit/nunit/pull/957#discussion_r43576230>:
>
> >          public void SummaryReportTest()
> >          {
> > -            var expected = new string[] {
> > +            var expected = new [] {
>
> I use implicit typed arrays throughout the CF code, but that, by
> definition, requires C# 3.0 (VS2008).
>
> So the question is, are we going to support any C# versions < 3.0?
> Software like mono and cecil condition for earlier frameworks and
> compilers. We could do that, but it would be messy. It would seem to be
> unnecessary in this case, since it is easy to just use explicit typed
> arrays. There is no difference in the generated code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nunit/nunit/pull/957/files#r43576230>.
>